src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -223,6 +238,41 @@ public class XdocsPagesTest {
     private static final Set<String> GOOGLE_MODULES = Collections.unmodifiableSet(
         CheckUtil.getConfigGoogleStyleModules());
 
+    private static final String XDOCS_ENCODING = "UTF-8";
+    private static final String XDOCS_HINT = "xdocs-template";
+
+    /**
+     * We want all xdocs to be generated from templates before validation.
+     *
+     * @throws Exception if something goes wrong
+     */
+    @Test
+    @Order(1)
+    public void parseAllTemplates() throws Exception {
+        final PlexusContainer plexus = new DefaultPlexusContainer();
+        final File userDir = new File(System.getProperty("user.dir"));
```

Comment:
Do we not have a constant for `user.dir`? Seems like we should, maybe in XDocUtil.

Reply:
Done. There are a few other usages of `user.dir`. Should I change those to use it as well or is it unrelated to this PR.
```
src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaWriter.java:        final String rootOutputPath = System.getProperty("user.dir") + "/src/main/resources";
src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateParser.java:                new File(System.getProperty("user.dir")));
src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java:    public static final String USER_DIR = System.getProperty("user.dir");
src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtilTest.java:        MetadataGeneratorUtil.generate(System.getProperty("user.dir")
src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtilTest.java:                Paths.get(System.getProperty("user.dir") + "/src/main/resources/com/puppycrawl"

```

Having it in `XdocUtil` doesn't allow me to use it in `XdocTemplateParser` thoguh.

Reply:
If there is a better/ more appropriate place to store it, I am good with that too. As far as other usages, we can just update them in another commit or send another PR with new constant and updates for other usages.